### PR TITLE
Upgrade dotnet 8

### PR DESCRIPTION
**Pull request by hahn-kev**
closing the previous PR in favor of this one as I pushed the branch to this repo instead of my fork

issue left over from previous PR
> I got all the tests building, however it looks like one of my workarounds is not working. There's a runtime error like this on a bunch of tests:
> 
> ```
> System.Reflection.ReflectionTypeLoadException : Unable to load one or more of the requested types.
> Could not load type 'SIL.UsbDrive.IUsbDriveInfo' from assembly 'SIL.Core.Desktop, Version=13.0.0.0, Culture=neutral, PublicKeyToken=cab3c8c5232dfcf2'.
>   ----> System.TypeLoadException : Could not load type 'SIL.UsbDrive.IUsbDriveInfo' from assembly 'SIL.Core.Desktop, Version=13.0.0.0, Culture=neutral, PublicKeyToken=cab3c8c5232dfcf2'.
>    at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
>    at System.Reflection.RuntimeModule.GetTypes()
>    at System.ComponentModel.Composition.Hosting.AssemblyCatalog.get_InnerCatalog()
>    at System.ComponentModel.Composition.Hosting.AssemblyCatalog.GetEnumerator()
>    at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
>    at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
>    at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
>    at System.ComponentModel.Composition.Primitives.ComposablePartCatalog.GetExports(ImportDefinition definition)
>    at System.ComponentModel.Composition.Hosting.CatalogExportProvider.InternalGetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.CatalogExportProvider.InnerCatalogExportProvider.GetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExports(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.CatalogExportProvider.GetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.GetExports(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.AggregateExportProvider.GetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExports(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.CompositionContainer.GetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.TryGetExportsCore(ImportDefinition definition, AtomicComposition atomicComposition, IEnumerable`1& exports)
>    at System.ComponentModel.Composition.Hosting.ExportProvider.GetExports(ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ImportEngine.TryGetExports(ExportProvider provider, ComposablePart part, ImportDefinition definition, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ImportEngine.TrySatisfyImportSubset(PartManager partManager, IEnumerable`1 imports, AtomicComposition atomicComposition)
>    at System.ComponentModel.Composition.Hosting.ImportEngine.TrySatisfyImportsStateMachine(PartManager partManager, ComposablePart part)
>    at System.ComponentModel.Composition.Hosting.ImportEngine.TrySatisfyImports(PartManager partManager, ComposablePart part, Boolean shouldTrackImports)
>    at System.ComponentModel.Composition.Hosting.ImportEngine.SatisfyImports(ComposablePart part)
>    at System.ComponentModel.Composition.Hosting.ComposablePartExportProvider.<>c__DisplayClass19_0.<Compose>b__0()
>    at System.ComponentModel.Composition.Hosting.CompositionServices.TryInvoke(Action action)
>    at System.ComponentModel.Composition.Hosting.ComposablePartExportProvider.Compose(CompositionBatch batch)
>    at System.ComponentModel.Composition.Hosting.CompositionContainer.Compose(CompositionBatch batch)
>    at System.ComponentModel.Composition.AttributedModelServices.ComposeParts(CompositionContainer container, Object[] attributedParts)
>    at Chorus.notes.EmbeddedMessageContentHandlerRepository..ctor()
>    at Chorus.ChorusUIComponentsInjector.Inject(ContainerBuilder builder, String projectPath, SyncUIFeatures syncDialogFeatures)
>    at Chorus.ChorusUIComponentsInjector.Inject(ContainerBuilder builder, String projectPath)
>    at WeSay.Project.WeSayWordsProject.RegisterChorusStuff(ContainerBuilder builder, ChorusNotesDisplaySettings displaySettings) in C:\dev\wesay\src\WeSay.Project\WeSayWordsProject.cs:line 676
>    at WeSay.Project.WeSayWordsProject.PopulateDIContainer(IProgressNotificationProvider progressProvider) in C:\dev\wesay\src\WeSay.Project\WeSayWordsProject.cs:line 552
>    at WeSay.Project.WeSayWordsProject.LoadFromProjectDirectoryPathInner(String projectDirectoryPath, IProgressNotificationProvider progressProvider) in C:\dev\wesay\src\WeSay.Project\WeSayWordsProject.cs:line 366
>    at WeSay.Project.WeSayWordsProject.LoadFromProjectDirectoryPath(String projectDirectoryPath, IProgressNotificationProvider progressProvider) in C:\dev\wesay\src\WeSay.Project\WeSayWordsProject.cs:line 336
>    at WeSay.Project.WeSayWordsProject.LoadFromLiftLexiconPath(String liftPath, IProgressNotificationProvider progressProvider) in C:\dev\wesay\src\WeSay.Project\WeSayWordsProject.cs:line 247
>    at WeSay.Project.Tests.ProjectDirectorySetupForTesting.CreateLoadedProject(IProgressNotificationProvider progressProvider) in C:\dev\wesay\src\WeSay.Project.Tests\ProjectDirectorySetupForTesting.cs:line 133
>    at WeSay.Project.Tests.ProjectDirectorySetupForTesting.CreateLoadedProject() in C:\dev\wesay\src\WeSay.Project.Tests\ProjectDirectorySetupForTesting.cs:line 128
>    at WeSay.Project.Tests.PLiftExporterTests.EnvironmentForTest..ctor() in C:\dev\wesay\src\WeSay.Project.Tests\PLiftExporterTests.cs:line 37
>    at WeSay.Project.Tests.PLiftExporterTests.Export_CitationHasVoice_CitationOmitsVoice() in C:\dev\wesay\src\WeSay.Project.Tests\PLiftExporterTests.cs:line 290
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
>    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
> ```
> 
> this is because SIL.Core.Desktop .netstandard is being used and the usb stuff is missing from that version. To fix this we need to produce a new version that includes that stuff.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/wesay/103)

### CODE REVIEW

**Review by josephmyers — APPROVED** [review]

File: build/build.mono.proj
Comment (megahirt) [inline build/build.mono.proj:1]: perhaps in a separate PR, we can remove all mono-specific references, like this file

File: src/LexicalModel.Tests/LexEntryRepositoryCachingTests.cs [hunk 1]
```diff
@@ -29,15 +29,16 @@ public void OneTimeTeardown()
...
+			_tempfolder = new TemporaryFolder("LexEntryRepositoryCachingTests");
+			string persistedFilePath = _tempfolder.GetNewTempFile(false).Path;
 			_repository = new LexEntryRepository(persistedFilePath);
 		}
 
 		[TearDown]
 		public void Teardown()
 		{
 			_repository.Dispose();
+			_repository.Dispose(); //added in dotnet8 conversion
```
Comment (megahirt) [inline src/LexicalModel.Tests/LexEntryRepositoryCachingTests.cs:41]: why do we have two of these in a row?  I'm guessing this should be `_tempfolder`

Comment (megahirt) [inline src/LexicalModel.Tests/LexEntryRepositoryCachingTests.cs:41]:
```suggestion
			_tempfolder.Dispose(); //added in dotnet8 conversion
```

**Review by megahirt — APPROVED** [review]
LGTM some non-blocking comments

**Comment by markusendres** [comment]
What is the current status of this? Is there help needed/wanted?
@megahirt & @josephmyers - can I work from this changeset up forward or are there any things you are currently working on yourself or don't want others to get involved?

**Comment by hahn-kev** [comment]
@markusendres Chris took some time to test out this version and ran into some issues, so we've decided not to pursue this PR.

@megahirt should we close this PR?

**Comment by markusendres** [comment]
> @markusendres Chris took some time to test out this version and ran into some issues, so we've decided not to pursue this PR.
> 
> @megahirt should we close this PR?

@hahn-kev : Thank you for your quick response.
Is it just the problem initially mentioned:
``
Could not load type 'SIL.UsbDrive.IUsbDriveInfo' from assembly 'SIL.Core.Desktop, Version=13.0.0.0, Culture=neutral, PublicKeyToken=cab3c8c5232dfcf2'.
  ----> System.TypeLoadException : Could not load type 'SIL.UsbDrive.IUsbDriveInfo' from assembly 'SIL.Core.Desktop, Version=13.0.0.0, Culture=neutral, PublicKeyToken=cab3c8c5232dfcf2'.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
``
or what are the other Problems?
Would be nice if we could document this so that the next person attempting to upgrade to .NET8 will be able to see what problems were blocking here.

**Comment by megahirt** [comment]
> Would be nice if we could document this so that the next person attempting to upgrade to .NET8 will be able to see what problems were blocking here.

@markusendres thank you.  I'll document it here.  This branch actually builds and runs as is.  I fired up WeSay and was able to open the program and click around and use it.  It mostly works but there are a number of UI issues of WinForms not behaving or showing up properly that we decided not to pursue this upgrade to `net8-windows`.  Must of the issues I found were in the config tool.  We have no plans to develop WeSay further and instead are investing in .net8 + web technology as we build a replacement.

We do however plan to ship and publish a version of WeSay that officially supports FieldWorks 9.  This will be based on .Net Framework 4.6 likely.

**Comment by markusendres** [comment]
Thank you @megahirt - is there already a repository for a webapp that would replace the functionality of WeSay? If so can you give me a hint on where to find it. If not, is this a process within SIL or is public contribution (outside of SIL) appreciated?

Is web-languageforge the repository? Or are there additional/other plans?

**Comment by markusendres** [comment]
Created #105 for the .NET Update to 4.6.2 or 4.8

**Comment by hahn-kev** [comment]
The replacement is being worked on here: https://github.com/sillsdev/languageforge-lexbox/tree/develop/backend%2FFwLite

**Comment by markusendres** [comment]
Thank you @hahn-kev

**Pull request closed without merging**